workos/utils/http_client.py

```diff
@@ -0,0 +1,173 @@
+import asyncio
+
+import httpx
+
+from workos.utils._base_http_client import BaseHTTPClient
+from workos.utils.request import REQUEST_METHOD_GET
+
+
+class SyncHttpxClientWrapper(httpx.Client):
+    def __del__(self) -> None:
+        try:
+            self.close()
+        except Exception:
+            pass
+
+
+class SyncHTTPClient(BaseHTTPClient[httpx.Client]):
+    """Sync HTTP Client for a convenient way to access the WorkOS feature set."""
+
+    _client: httpx.Client
+
+    def __init__(
+        self,
+        *,
+        base_url: str,
+        version: str,
+        timeout: int = None,
+        transport: httpx.BaseTransport = httpx.HTTPTransport(),
+    ) -> None:
+        super().__init__(
+            base_url=base_url,
+            version=version,
+            timeout=timeout,
+        )
+        self._client = SyncHttpxClientWrapper(
+            base_url=base_url,
+            timeout=timeout,
+            follow_redirects=True,
+            transport=transport,
+        )
+
+    def is_closed(self) -> bool:
+        return self._client.is_closed
+
+    def close(self) -> None:
```

Comment:
When does this get used?

Reply:
Took this from Stainless. We don't use close directly, but it is used if the `__exit__` method is called by the Python runtime.  This is mostly a helper if we want to cleanup on a fatal error.